packages/protocol/test/team/TimelockTokenPool.t.sol

```diff
@@ -136,62 +178,80 @@ contract TestTimelockTokenPool is TaikoTest {
         uint64 grantCliff = grantStart + grantPeriod / 2;
 
         pool.grant(
-            Alice, TimelockTokenPool.Grant(10_000e18, grantStart, grantCliff, grantPeriod, 0, 0, 0)
+            Alice,
+            TimelockTokenPool.Grant(
+                10_000e18, strikePrice1, grantStart, grantCliff, grantPeriod, 0, 0, 0
+            )
         );
         vm.prank(Vault);
         tko.approve(address(pool), 10_000e18);
 
+        vm.prank(Alice);
+        usdc.approve(address(pool), 10_000e18 / ONE_TKO_UNIT * strikePrice1);
+
         (
             uint128 amountOwned,
             uint128 amountUnlocked,
             uint128 amountWithdrawn,
-            uint128 amountWithdrawable
+            uint128 amountToWithdraw,
+            uint128 costToWithdraw
         ) = pool.getMyGrantSummary(Alice);
         assertEq(amountOwned, 0);
         assertEq(amountUnlocked, 0);
         assertEq(amountWithdrawn, 0);
-        assertEq(amountWithdrawable, 0);
+        assertEq(amountToWithdraw, 0);
+        assertEq(costToWithdraw, 0);
 
         vm.warp(grantCliff);
 
-        (amountOwned, amountUnlocked, amountWithdrawn, amountWithdrawable) =
+        (amountOwned, amountUnlocked, amountWithdrawn, amountToWithdraw, costToWithdraw) =
             pool.getMyGrantSummary(Alice);
         assertEq(amountOwned, 0);
         assertEq(amountUnlocked, 0);
         assertEq(amountWithdrawn, 0);
-        assertEq(amountWithdrawable, 0);
+        assertEq(amountToWithdraw, 0);
+        assertEq(costToWithdraw, 0);
 
         vm.warp(grantCliff + 1);
 
-        (amountOwned, amountUnlocked, amountWithdrawn, amountWithdrawable) =
+        (amountOwned, amountUnlocked, amountWithdrawn, amountToWithdraw, costToWithdraw) =
             pool.getMyGrantSummary(Alice);
+
         uint256 amount1 = 5_000_000_317_097_919_837_645;
+        uint256 expectedCost = amount1 / ONE_TKO_UNIT * strikePrice1;
+
         assertEq(amountOwned, amount1);
         assertEq(amountUnlocked, amount1);
         assertEq(amountWithdrawn, 0);
-        assertEq(amountWithdrawable, amount1);
+        assertEq(amountToWithdraw, amount1);
+        assertEq(costToWithdraw, expectedCost);
 
         vm.prank(Alice);
         pool.withdraw();
 
         vm.warp(grantStart + grantPeriod + 365 days);
 
-        (amountOwned, amountUnlocked, amountWithdrawn, amountWithdrawable) =
+        (amountOwned, amountUnlocked, amountWithdrawn, amountToWithdraw, costToWithdraw) =
             pool.getMyGrantSummary(Alice);
+
+        expectedCost = (10_000e18 - amount1) / ONE_TKO_UNIT * strikePrice1;
         assertEq(amountOwned, 10_000e18);
         assertEq(amountUnlocked, 10_000e18);
         assertEq(amountWithdrawn, amount1);
-        assertEq(amountWithdrawable, 10_000e18 - amount1);
+        assertEq(amountToWithdraw, 10_000e18 - amount1);
+        // TODO(dani): the following assert fails
```

Comment:
@adaki2004 Please take a look at this TODO - uncommenting the next line will fail the test. I'm not sure it's a test issue or implementation issue.

Reply:
Checked quikcly - no issue on any end, it is a rounding error. Will check tomo where exactly that loss.
```
  expected cost: 49990000
  costToWithdraw: 50000000
```